plugins/lookup/dig.py

```diff
@@ -358,7 +363,6 @@ def run(self, terms, variables=None, **kwargs):
                             nameservers.append(nsaddr)
                         except Exception as e:
                             raise AnsibleError("dns lookup NS: %s" % to_native(e))
-                    myres.nameservers = nameservers
```

Comment:
The old code only keeps the nameservers from the last entry in `@`, while with your change all nameservers provided by the user are used.

I would say this is a bugfix. (Or is `myres.nameservers = ...` doing some magic so that if you do it twice, the list of nameservers is extended?)

Do you mind moving the nameserver modification to a separate PR? Bugfixes are backported to more stable branches than new features (which adding `port` is).